### PR TITLE
Add Create Plan Feature

### PR DESCRIPTION
**Pull request by Musfirahe0556596**
A plan object consists of a plan name and a list of workouts. The hashmap implementation (mainly in PlanList.java) is similar to the Workout's one. That is, HashMap<String, Plan>, means that the unique Key is the Plan name and the value is the Plan object.

### CODE REVIEW

**Review by a1021492980 — APPROVED** [review]
Good and quick implementation of Creating Plan feature! Thank you and your hard work will be remembered!

**Comment by Musfirahe0556596** [comment]
> Good and quick implementation of Creating Plan feature! Thank you and your hard work will be remembered!

Thank you :D!

File: src/main/java/commands/PlanCommand.java [hunk 1]
```diff
@@ -1,11 +1,145 @@
...
+     */
+    public void setUserAction(String userAction) throws InvalidCommandException {
+        switch (userAction) {
+        case CREATE_ACTION_KEYWORD:
+        case LIST_ACTION_KEYWORD:
+        case DELETE_ACTION_KEYWORD:
+        case DETAILS_ACTION_KEYWORD:
+            this.userAction = userAction;
+            break;
+        default:
```
Comment (TianaiYan) [inline src/main/java/commands/PlanCommand.java:89]: Probably a //Fallthrough comment should be added 
From the [Java coding standard](https://se-education.org/guides/conventions/java/basic.html#layout) website:
The explicit `//Fallthrough` comment should be included whenever there is a case statement without a break statement.

**Review by TianaiYan — COMMENTED** [review]
Looks great! Thank you.

File: src/main/java/commands/PlanCommand.java [hunk 2]
```diff
@@ -1,11 +1,145 @@
...
+    private static Logger logger = Logger.getLogger(WorkoutCommand.class.getName());
+
+    public PlanCommand(String userInput, FileManager fileManager, PlanList planList,
+                       String userAction, String userArguments) throws InvalidCommandException {
         super(userInput);
+        this.fileManager = fileManager;
+        this.planList = planList;
+        setUserAction(userAction);
+        this.userArguments = userArguments;
     }
```
Comment (alanlowzies) [inline src/main/java/commands/PlanCommand.java:36]: Should include `LogHandler.linkToFileLogger(logger);` to get the log messages in this class to be written to the log file.

File: src/main/java/commands/PlanCommand.java [hunk 3]
```diff
@@ -1,11 +1,145 @@
...
+            System.out.println(e.getMessage());
+            System.out.println("Please try again");
+        } catch (InvalidPlanException e) {
+            System.out.println(e.getMessage());
+            System.out.println("Please try again");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Uh oh, it seems like too few arguments were entered.");
+            System.out.println("Please try again. Alternatively, type 'help' if you need\n"
+                    + "more information on the commands.");
+
```
Comment (alanlowzies) [inline src/main/java/commands/PlanCommand.java:134]: Perhaps this empty line can be removed to be consistent with the other `catch` blocks?

File: src/main/java/commands/PlanCommand.java [hunk 4]
```diff
@@ -1,11 +1,145 @@
...
+        } catch (NumberFormatException e) {
+            logger.log(Level.WARNING, "A non-formattable number was received!");
+            System.out.println("Uh oh, a number was expected in your input, but a non-formattable\n"
+                    + "number was received.");
+            System.out.println("Please try again. Alternatively, type 'help' if you need\n"
+                    + "more information on the commands.");
+        }
```
Comment (alanlowzies) [inline src/main/java/commands/PlanCommand.java:142]: Perhaps this extra line can be removed?

File: src/main/java/data/plans/Plan.java [hunk 5]
```diff
@@ -1,4 +1,36 @@
 package data.plans;
 
+import data.workouts.Workout;
+
+import java.util.ArrayList;
+
 public class Plan {
+    private String planName;
+    private ArrayList<Workout> workoutsInPlanList = new ArrayList<>();
```
Comment (alanlowzies) [inline src/main/java/data/plans/Plan.java:9]: Perhaps this variable name can be renamed to `listOfWorkouts`?

File: src/main/java/data/plans/PlanList.java [hunk 6]
```diff
@@ -1,4 +1,140 @@
...
+
+    public ArrayList<String> getPlansDisplayList() {
+        return this.plansDisplayList;
+    }
+
+    public HashMap<String, Plan> getPlansHashMapList() {
+        return this.plansHashMapList;
+    }
+
+    public static final int MAX_NUMBER_OF_WORKOUTS_IN_A_PLAN = 10;
```
Comment (alanlowzies) [inline src/main/java/data/plans/PlanList.java:27]: Perhaps this constant can be moved to the top of the class so as to maintain consistency with the other classes.

**Review by alanlowzies — COMMENTED** [review]
Looks alright. Just some coding standard suggestions and the Logger thing. Thank you

File: src/main/java/commands/PlanCommand.java (hunk 1 above)
Comment (Musfirahe0556596) [reply src/main/java/commands/PlanCommand.java:89]: thank you!

File: src/main/java/commands/PlanCommand.java (hunk 2 above)
Comment (Musfirahe0556596) [reply src/main/java/commands/PlanCommand.java:36]: added!

**Pull request merged**